pkg/registry/core/pod/strategy.go

```diff
@@ -810,13 +811,24 @@ func applyAppArmorVersionSkew(pod *api.Pod) {
 					// Skip copying invalid value.
 					newField = nil
 				}
+				deprecationWarning := false
 
 				// Only copy the annotation to the field if it is different from the pod-level profile.
 				if newField != nil && !apiequality.Semantic.DeepEqual(newField, podProfile) {
 					if ctr.SecurityContext == nil {
 						ctr.SecurityContext = &api.SecurityContext{}
 					}
 					ctr.SecurityContext.AppArmorProfile = newField
+					deprecationWarning = true
+				}
+
+				// Warn even on invalid annotations (newField == nil).
+				if (deprecationWarning || newField == nil) && utilfeature.DefaultFeatureGate.Enabled(features.AppArmorFields) {
```

Comment:
no need to check AppArmorFields enablement, we check that at the top of the method